account_analytic_spread_by_tag/models/account_move_line.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2024 APSL-Nagarro - Miquel Alzanillas
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+from odoo import models
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    def _search_analytic_accounts_by_tag(self, analytic_tag_id):
+        AccountAnalyticAccount = self.env["account.analytic.account"]
+        account_analytic_acc_in_filter = (
+            analytic_tag_id.spread_filter_analytic_account_ids
+        )
+        filter_operation = analytic_tag_id.spread_filter_operation
+
+        if filter_operation and filter_operation == "exclude":
+            return AccountAnalyticAccount.search(
+                [
+                    ("mapped_analytic_tag_ids", "in", [analytic_tag_id.id]),
+                    ("id", "not in", account_analytic_acc_in_filter.ids),
+                ]
+            )
+
+        if filter_operation and filter_operation == "include":
+            return AccountAnalyticAccount.search(
+                [
+                    ("mapped_analytic_tag_ids", "in", [analytic_tag_id.id]),
+                    ("id", "in", account_analytic_acc_in_filter.ids),
+                ]
+            )
+        else:
+            return AccountAnalyticAccount.search(
+                [("mapped_analytic_tag_ids", "in", [analytic_tag_id.id])]
+            )
+
+    def _prepare_analytic_lines(self):
+        """Check if any of the tags have spread option enabled"""
+        vals = super()._prepare_analytic_lines()
+        tag_ids_to_spread = self.analytic_tag_ids.filtered(lambda x: x.to_spread)
+        if tag_ids_to_spread:
+            for tag in tag_ids_to_spread:
+                account_analytic_ids = self._search_analytic_accounts_by_tag(tag)
+                number_of_aac = account_analytic_ids.__len__()
+                if number_of_aac > 0:
+                    line_amount = self.balance / number_of_aac
+                    for aac in account_analytic_ids:
+                        default_name = self.name or (
+                            self.ref
+                            or "/"
+                            + " -- "
+                            + (self.partner_id and self.partner_id.name or "/")
+                        )
+                        vals.append(
+                            {
+                                "name": default_name,
+                                "date": self.date,
+                                "account_id": aac.id,
```

Comment:
This approach doesn't work correctly because it always assigns the analytic account to the project plan defined in the settings:
```python
self.env["ir.config_parameter"].sudo().get_param("analytic.project_plan", 0) 
```
This retrieves the plan that should be in the account_id.

You need to determine whether the plan belongs in the account_id or in one of the dynamic x_plan{id}_id fields. These x_plan fields are autogenerated based on the analytic plans you have, identified by their ID.

To fix this, you can do the following:
```suggestion
                                aac.root_plan_id._column_name(): aac.id,
```

Reply:
Maybe @miquelalzanillas can review the changes and then open a FIX pr for this.

Reply:
Thanks for the review @BernatObrador, @miquelalzanillas what do you think about it?

Reply:
Hi @peluko00 , 

@BernatObrador is right.

Analytic plan definition is needed here.

Reply:
Done, thanks! Can you review again @miquelalzanillas @BernatObrador please?